qiskit_nature/drivers/molecule.py

```diff
@@ -84,6 +90,40 @@ def __str__(self) -> str:
                 string += [f"\t\t{atom}\t{mass}"]
         return "\n".join(string)
 
+    @staticmethod
+    def convert_units(molecule: Molecule, unit: UnitsType) -> Molecule:
```

Comment:
I am wondering about the need/use of this method. This alters the geometry but if there are degrees of freedom setup with the molecule then these remain how they were. Would it be better/simpler, if this conversion is really needed/wanted to just return the base geometry in the supplied units and let the caller build out a new Molecule. I wonder in practice how much this would be used - for other driver inputs we let the user specify either way and its up to them to do the conversion themselves. This could be left that way so the user supplies the molecule in either bohr or angstrom and if they want to convert from one set of units to the other we let them do it themselves.

Reply:
Hm yes I did not think about the degrees of freedom.. I guess the mathematics are pretty straight forward anyways..
Would you rather we remove this entire conversion capability then or would you still provide it but instead of returning `Molecule` it just returns the new geometry?

Reply:
My personal choice would be to remove it entirely. A user can work in whatever units they want and if they have something that's needs conversion it easy enough for them to do it - they have to do it for input to other drivers. If one really wanted to supply conversion functions one could do that in with the constants - not that I am proposing to do that since again its straightforward to do.